book/styles.md

```diff
@@ -612,85 +648,128 @@ def relative_url(url, current):
             url = url[3:]
-        return current + "/" + url
+        return dir + "/" + url
 ```
 
-In the second case, the `[:3]` and the `"/".join` handle the two
-slashes that come after `http:` in the URL, while in the last case,
-the logic ensures that a link to `foo.html` on `http://a.com/bar.html`
-goes to `http://a.com/foo.html`, not `http://a.com/bar.html/foo.html`.
-
 Let's put it all together. We want to collect CSS rules from each of
 the linked files, and the browser style sheet, into one big list so we
 can apply each of them. So let's add them onto the end of the `rules`
 list:
 
-``` {.python}
+``` {.python indent=4}
 def load(self, url):
     # ...
-    for link in find_links(nodes, []):
-        header, body = request(relative_url(link, url))
-        rules.extend(CSSParser(body).parse())
+    for link in links:
+        try:
+            header, body = request(relative_url(link, url))
+            rules.extend(CSSParser(body).parse())
+        except:
+            continue
 ```
 
+The `try`/`except` here handles the case where downloading a style
+sheet fails---in which case the browser just ignores it. But if you're
+debugging `relative_url` try removing the `try`/`except` here, which
+can hide errors.
+
 Since the page's stylesheets come *after* browser style, user styles
-take priority over the browser style sheet.^[In reality this is
-handled by browser styles having a lower score than user styles in the
-cascade order, but our browser style sheet only has tag selectors in
-it, so every rule already has the lowest possible score.] With the
-rules loaded, we need only sort and apply them and then do layout:
+take priority over the browser style sheet. With the rules loaded, we
+need only sort and apply them and then do layout:
 
-``` {.python}
+``` {.python indent=4}
 def load(self, url):
     # ...
     rules.sort(key=lambda x: x[0].priority())
     rules.reverse()
     style(nodes, rules)
-    self.layout(nodes)
+    # ...
 ```
 
-With this done, each page should now automatically apply the margins
-and paddings specified in the browser stylesheet, making it possible
-to delete some of `InlineLayout`'s tag handlers in its `close` method.
+Now every element is styled and ready to be laid out and drawn to the
+screen. Open this page up again, and you should see both gray
+backgrounds on every code block (thanks to the browser style sheet)
+and light-gold backgrounds on this book's mailing list signup form
+(try the book's main page).
+
+Alright: we've got background colors that can be configured by web
+page authors. But there's more to web design than that! At the very
+least, if you're changing background colors you might want to change
+foreground colors as well---the CSS `color` property. For example,
+usually links are blue. But there's a catch: `color` affects text, but
+text nodes don't have any styles at all. How can that work?
+
+::: {.further}
+Each browser has its own browser style sheet ([Chrome][blink-css],
```

Comment:
```suggestion
Each browser has its own browser style sheet ([Chromium][blink-css],
```

Reply:
I'm guessing there's some convention to this. I avoided saying Blink, Webkit, Gecko here because those terms aren't as widely known, even though it's more accurate. Is that better? Is there some legal guideline on using "Chrome"?

Reply:
Not a legal guideline. I don't want to say Chrome because it implies it's the feature is not in Edge, Brave, Opera, etc.

I'd be happy with just changing Chrome to Chromium.